ui/src/app/applications/components/application-create-panel/application-create-panel.tsx

```diff
@@ -176,80 +177,55 @@ export const ApplicationCreatePanel = (props: {
+                                                    type: 'Directory',
+                                                    details: {},
+                                                }));
+                                            } else {
+                                                return {
+                                                    type: 'Directory',
+                                                    details: {},
+                                                };
+                                            }
+                                        }}>
+                                    {(details: models.RepoAppDetails) => {
+                                        const type = explicitPathType && explicitPathType.path === app.spec.source.path && explicitPathType.type || details.type;
+                                        if (details.type !== type) {
+                                            switch (type) {
+                                                case 'Helm':
+                                                    details = {type, path: details.path, helm: { name: '', valueFiles: [],  path: '', parameters: [] } };
+                                                    break;
+                                                case 'Kustomize':
+                                                    details = {type, path: details.path, kustomize: { path: '' } };
+                                                    break;
+                                                case 'Ksonnet':
+                                                    details = {type, path: details.path, ksonnet: { name: '', path: '', environments: {}, parameters: []} };
+                                                    break;
+                                                // Directory or Plugin
+                                                default:
+                                                    details = {type, path: details.path, directory: {} };
+                                                    break;
+                                            }
+                                        }
+                                        return (
+                                            <React.Fragment>
+                                                <DropDownMenu anchor={() => (<p>{type} <i className='fa fa-caret-down'/></p>)} items={appTypes.map(
+                                                    (item) => ({ title: item.type, action: () => {
+                                                        setExplicitPathType({ type: item.type, path: app.spec.source.path });
+                                                        normalizeTypeFields(api, item.type);
+                                                    }}))}
+                                                />
+                                                <ApplicationParameters noReadonlyMode={true} application={app} details={details} save={async (updatedApp) => {
```

Comment:
I would prefer to keep readonly enabled by default, because this is the most common use case. In edge cases like application creation panel readonly can be disabled using this parameter